include/sunheuristics/sunheuristics_default.h

```diff
@@ -0,0 +1,120 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2023, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the header file for the SUNHeuristics_Default module.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNHEURISTICS_DEFAULT_H
+#define _SUNHEURISTICS_DEFAULT_H
+
+#include <stdio.h>
+#include <sundials/sundials_heuristics.h>
+
+#ifdef __cplusplus  /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/* ---------------------------------------
+ * Default implementation of SUNHeuristics
+ * --------------------------------------- */
+
+struct _SUNHeuristicsContent_Default {
+  realtype     hmax_inv;   /* inverse of maximum allowable time step     */
+  realtype     hmin;       /* minimum allowable time step                */
+  realtype     etamax;     /* eta <= etamax                              */
+  realtype     etamx1;     /* max step size change on first step         */
+  realtype     etamxf;     /* h reduction factor on multiple error fails */
+  realtype     etamin;     /* eta >= etamin on error test fail           */
+  int          small_nef;  /* bound to determine 'multiple' above        */
+  realtype     etacf;      /* h reduction factor on nonlinear conv fail  */
+  SUNExpStabFn expstab;    /* step stability function                    */
+  void*        estab_data; /* user pointer passed to expstab             */
+  realtype     cfl;        /* cfl safety factor                          */
+  realtype     safety;     /* step safety factor                         */
```

Comment:
The heuristics object handles the safety factor, but the controller handles a bias. It is a bit strange they are in different classes when they are so similar.

Reply:
I agree that this separation seems artificial at first.  I initially had both inside the adaptivity controller, but moved this safety factor here to sit alongside the corresponding CFL stability safety factor.  The main idea was that the controller specifies the step it recommends to meet the error threshold, but any post-processing of that recommendation (e.g., via `safety`) constitutes an heuristic change.

Furthermore, although both `bias` and `safety` play similar roles, `bias` multiplies the local temporal error estimate directly, whereas `safety` sits outside the `1/(p+1)` exponent.  Thus, I consider the `bias` as encoding a user-provided estimate of how a given problem will relate local to global error (or for that matter, how much more tightly the user wants local error to be controlled in the hopes of achieving a given global error).  

Hence the separation of `bias` and `safety` in this PR.